databricks/koalas/indexes.py

```diff
@@ -2703,27 +2694,24 @@ def _is_monotonic_increasing(self):
 
     @staticmethod
     def _comparator_for_monotonic_decreasing(data_type):
-        if isinstance(data_type, StringType):
-            return compare_disallow_null
-        elif isinstance(data_type, BooleanType):
-            return compare_allow_null
-        elif isinstance(data_type, NumericType):
-            return compare_null_last
-        else:
-            return compare_null_first
+        return compare_disallow_null
 
     def _is_monotonic_decreasing(self):
         scol = self.spark.column
         window = Window.orderBy(NATURAL_ORDER_COLUMN_NAME).rowsBetween(-1, -1)
         prev = F.lag(scol, 1).over(window)
 
         cond = F.lit(True)
+        has_not_null = F.lit(True)
         for field in self.spark.data_type[::-1]:
             left = scol.getField(field.name)
             right = prev.getField(field.name)
             compare = MultiIndex._comparator_for_monotonic_decreasing(field.dataType)
+            # Since pandas 1.1.4, null value is not allowed at any levels of MultiIndex.
+            # Therefore, we should check `has_not_null` over the all levels.
+            has_not_null = has_not_null & left.isNotNull() & right.isNotNull()
             cond = F.when(left.eqNullSafe(right), cond).otherwise(
-                compare(left, right, spark.Column.__lt__)
+                has_not_null & compare(left, right, spark.Column.__lt__)
```

Comment:
I'm not quite sure about this part.

Can't we just use `has_not_null` later? e.g.,

```py
cond = prev.isNull() | (has_not_null & cond)
```

Reply:
Thanks for the opinion!
Unfortunately, then we cannot cover a case when `data=[(None, None), (None, None), (None, None), (None, None), (None, None)]`  😿

Reply:
What's the expected result for the case?

Reply:
```py
>>> kdf = ks.DataFrame({'a': ['a', 'b', 'c'], 'b': [1, 2, 3]})
>>> kdf['a'] = None
>>> kdf['b'] = None
>>> kmidx = kdf.set_index(['a', 'b']).index
>>> kmidx
MultiIndex([(nan, nan),
            (nan, nan),
            (nan, nan)],
           names=['a', 'b'])
>>> kmidx.is_monotonic_increasing
True
>>> kmidx.to_pandas().is_monotonic_increasing
False
```

Reply:
Oh...... sorry I missed that case since current test skip this for pandas >= 1.1.0.

Will fix the tests and make it work properly. thanks!!